packages/block-editor/src/components/block-visibility/modal.js

```diff
@@ -0,0 +1,286 @@
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { useEffect, useState, useMemo } from '@wordpress/element';
+import {
+	Button,
+	CheckboxControl,
+	Flex,
+	FlexItem,
+	Icon,
+	Modal,
+} from '@wordpress/components';
+import { desktop, tablet, mobile, seen, unseen } from '@wordpress/icons';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { store as keyboardShortcutsStore } from '@wordpress/keyboard-shortcuts';
+
+/**
+ * Internal dependencies
+ */
+import useBlockDisplayInformation from '../use-block-display-information';
+import { store as blockEditorStore } from '../../store';
+import { cleanEmptyObject } from '../../hooks/utils';
+
+const VIEWPORTS = [
+	{ key: 'desktop', label: __( 'Hide on desktop' ), icon: desktop },
+	{ key: 'tablet', label: __( 'Hide on tablet' ), icon: tablet },
+	{ key: 'mobile', label: __( 'Hide on mobile' ), icon: mobile },
```

Comment:
Looks like we could generate those based on `VIEWPORT_LABELS`? Especially if we kept the key lowercase.